sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Util.scala

```diff
@@ -295,18 +295,61 @@ private[sql] object CatalogV2Util {
     catalog.name().equalsIgnoreCase(CatalogManager.SESSION_CATALOG_NAME)
   }
 
-  def convertTableProperties(
+  def convertTableProperties(c: CreateTableStatement): Map[String, String] = {
+    convertTableProperties(
+      c.properties, c.options, c.serde, c.location, c.comment, c.provider, c.external)
+  }
+
+  def convertTableProperties(c: CreateTableAsSelectStatement): Map[String, String] = {
+    convertTableProperties(
+      c.properties, c.options, c.serde, c.location, c.comment, c.provider, c.external)
+  }
+
+  def convertTableProperties(r: ReplaceTableStatement): Map[String, String] = {
+    convertTableProperties(r.properties, r.options, r.serde, r.location, r.comment, r.provider)
+  }
+
+  def convertTableProperties(r: ReplaceTableAsSelectStatement): Map[String, String] = {
+    convertTableProperties(r.properties, r.options, r.serde, r.location, r.comment, r.provider)
+  }
+
+  private def convertTableProperties(
       properties: Map[String, String],
       options: Map[String, String],
+      serdeInfo: Option[SerdeInfo],
       location: Option[String],
       comment: Option[String],
-      provider: Option[String]): Map[String, String] = {
-    properties ++ options ++
+      provider: Option[String],
+      external: Boolean = false): Map[String, String] = {
+    properties ++
+      options ++ // to make the transition to the "option." prefix easier, add both
+      options.map { case (key, value) => TableCatalog.OPTION_PREFIX + key -> value } ++
+      convertToProperties(serdeInfo) ++
+      (if (external) Map(TableCatalog.PROP_EXTERNAL -> "true") else Map.empty) ++
```

Comment:
The change might be small in line of code, but they are API/behavior changes that need individual JIRA tickets.

Reply:
Please be more specific about what you want changed.

Reply:
It's better if you can pull them out and make separated PRs. At least we need to create JIRA tickets for them, and put them into the PR title. Changes that need JIRA tickets:
1. Convert serde info to table properties for v2 create plans.
2. Add an `option` namespace to distinguish SERDEPROPERTIES and OPTIONS in the table properties.
3. Add the `EXTERNAL` table property to indicate CREATE EXTERNAL TABLE.

Reply:
I'm not sure Separate PR's really makes sense since this seems to be coherent whole to me, but if we need more Jira's just for change tracking that seems fine although I'm not sure how much use that is? Is the idea that it will be easier to use JIRA search to find when the changes occurred because I usually find tracking back from the commit to the Jira much easier than going from Jira -> the code. We also rarely have Jira's as updated as the PR's and commits.

Reply:
Making separate JIRAs and PRs:

- make it easier to port/revert/manage, e.g.) reverting a specific change.
- easier to track the discussions specific to smaller scope, e.g.) the current discussion thread looks already very long to read
- +1653 and −1125 change is large. It makes easily people to miss key things during a review.
- this `CREATE TABLE` syntax is pretty much an entry point of users. Here is where we need more calls for review. I would encourage to make it easier to review so many people can review closely to prevent any surprise by simply overlooking.

If I am not mistaken, it is usually considered as a good practice to split if that can logically be split. I also encourage to split PRs in general. If we should not split, I think the argument should be about why it should not be split (instead of why it should be split).

Reply:
@RussellSpitzer We use JIRA tickets to generate release notes. It's hard for the release manager to tell the API/behavior changes I mentioned earlier from this PR title "Unify create table syntax".

Also as @HyukjinKwon mentioned, smaller PRs can help to narrow down the discussion scope. This PR is hard for tracking purposes as the discussion is so long. Splitting this PR won't help too much now as the discussion is already done, but it can still help code review.

My baseline is to create separated JIRA tickets, to mention the decisions explicitly. For example, in this PR, `OPTIONS ...` will appear twice in the table properties, once with `option.` prefix, once without.

Reply:
I mentioned I don't have a problem with additional Jiras although I still probably won't trust Jira :), but I still don't see how the multiple PR's help, rather than talking about the generic case let's try to be specific why this pr needs to be broken up or what sections need more discussion and find folks to review them. 

I think the argument for keeping everything together is pretty strong. 

* These API changes are so heavily coupled with the rest of the ticket and any discussion of them without the context of the code changes here doesn't make sense. 
* All of the changes serve the same purpose and accomplish that purpose in the same way, (By passing formerly parsed options as table options)
* Because of the heavy coupling a maintainer will still have to treat the various changes as a single group so it will be difficult to revert/maintain the changes if they are isolated to different PR's.
* At best, if we separate the changes we only remove a small amount of code from this PR only to have code in separate PRs that doesn't make sense without this pr

One thing I am particularly worried about here is that we are quickly drifting into meta-discussion rather than talking about the actual code. This PR has been in review for 7 months and in discussion for longer than that. To be constructive I believe we should focus our efforts on actual changes to the code or why or why not we think various API changes should be made. 

I think if we want to help with code review we really just need to find more dedicated reviewers for the ticket, I think that would probably be much more productive than splitting the PR into pieces.

Reply:
Different people may have a different understanding of when it's necessary to split a PR. I don't have a big problem with it. Again I was only asking for individual JIRA tickets. One idea is to make an umbrella JIRA with 4 sub-tasks:
- Have a unified CREATE TABLE syntax that supports v1/v2 catalog
  - sub-task 1: merge the native and Hive-style CREATE TABLE SQL syntax (need to explain the merged syntax, e.g. USING and STORED AS/BY can't coexist, PARTITION BY column name with data type and PARTITION BY transforms can't coexist)
  - sub-task 2: convert serde info to v2 table properties (explain the rules)
  - sub-task 3: add EXTERNAL table property to indicate CREATE EXTERNAL TABLE
  - sub-task 4: add `options.` prefix for table properties specified in `OPTIONS ...` (explain that we duplicate `OPTIONS ...` in table properties without `option.` prefix for backward compatibility. this probably need some discussion.)

Personally I don't really see how sub-task 4 is related to CREATE TABLE unification. Maybe we should put it as a separated JIRA ticket.

Anyway I'll start detailed code review this week.

Reply:
The `option.` prefix is added in this PR because the serde properties are new in this PR. Those are stored in the same place as the options clause, so we need to fix how both are handled. The solution is compatible with the existing behavior because the options clause properties are added with and without the prefix. The serde properties are only added with the prefix.